common/json_stream.c

```diff
@@ -485,6 +484,16 @@ void json_add_short_channel_id(struct json_stream *response,
 			 short_channel_id_outnum(scid));
 }
 
+void json_add_short_channel_id_dir(struct json_stream *response,
+			       const char *fieldname,
+			       struct short_channel_id_dir scidd)
```

Comment:
scidd is a complex type, so unlike scids it should be passed as "const struct short_channel_id_dir *".  This also means the definition doesn't need to be in the header...